tools/accuracy_checker/accuracy_checker/metrics/im2latex_images_match.py

```diff
@@ -0,0 +1,358 @@
+template = r"""
+\documentclass[12pt]{article}
+\pagestyle{empty}
+\usepackage{amsmath}
+\newcommand{\mymatrix}[1]{\begin{matrix}#1\end{matrix}}
+\newcommand{\mypmatrix}[1]{\begin{pmatrix}#1\end{pmatrix}}
+\begin{document}
+\begin{displaymath}
+%s
+\end{displaymath}
+\end{document}
+"""
+
+
+def crop_image(img, output_path, default_size=None):
+    old_im = cv.imread(img, cv.IMREAD_GRAYSCALE)
+    img_data = np.copy(old_im)
+    nnz_inds = np.where(img_data != 255)
+    if len(nnz_inds[0]) == 0:
+        if not default_size:
+            cv.imwrite(output_path, old_im)
+            return False
+        assert len(default_size) == 2, default_size
+        x_min, y_min, x_max, y_max = 0, 0, default_size[0], default_size[1]
+        old_im = old_im[y_min: y_max + 1, x_min, x_max + 1]
+        cv.imwrite(output_path, old_im)
+        return False
+    y_min = np.min(nnz_inds[0])
+    y_max = np.max(nnz_inds[0])
+    x_min = np.min(nnz_inds[1])
+    x_max = np.max(nnz_inds[1])
+
+    old_im = old_im[y_min: y_max + 1, x_min: x_max + 1]
+    cv.imwrite(output_path, old_im)
+    return True
+
+
+def run(cmd, timeout_sec):
+    proc = subprocess.Popen(cmd, shell=True)
+
+    def kill_proc(process):
+        return process.kill()
+    timer = Timer(timeout_sec, kill_proc, [proc])
+    try:
+        timer.start()
+        _, _ = proc.communicate()
+    finally:
+        timer.cancel()
+
+
+def preprocess_formula(l):
+    """Formula preprocessing
+
+    Args:
+        l (str): Input formula
+
+    Returns:
+        str: Preprocessed formula
+    """
+    l = l.strip()
+    l = l.replace(r'\pmatrix', r'\mypmatrix')
+    l = l.replace(r'\matrix', r'\mymatrix')
+    # remove leading comments
+    l = l.strip('%')
+    if len(l) == 0:
+        l = '\\hspace{1cm}'
+    # \hspace {1 . 5 cm} -> \hspace {1.5cm}
+    for space in ["hspace", "vspace"]:
+        match = re.finditer(space + " {(.*?)}", l)
+        if match:
+            new_l = ""
+            last = 0
+            for m in match:
+                new_l = new_l + l[last:m.start(1)] + m.group(1).replace(" ", "")
+                last = m.end(1)
+            new_l = new_l + l[last:]
+            l = new_l
+    return l
+
+
+def render_routine(line):
+    """Function for rendering single formula
+
+    Args:
+        line (tuple): formula idx, formula string, path to store rendered image
+    """
+    formula, file_idx, folder_path = line
+    output_path = os.path.join(folder_path, file_idx)
+    pre_name = output_path.replace('/', '_').replace('.', '_')
+    formula = preprocess_formula(formula)
+    if not os.path.exists(output_path):
+        tex_filename = pre_name + '.tex'
+        log_filename = pre_name + '.log'
+        aux_filename = pre_name + '.aux'
+        with open(tex_filename, "w") as w:
+            w.write(template % formula)
+        run("pdflatex -interaction=nonstopmode {}  >/dev/null".format(tex_filename), TIMEOUT)
+        os.remove(tex_filename)
```

Comment:
please add checks for files existence to avoid attempting to delete non-existent files

```suggestion
        if os.path.exists(tex_filename):
            os.remove(tex_filename)
```